apps/dot-voting/test/dot-voting.test.js

```diff
@@ -342,7 +397,71 @@ contract('DotVoting App', accounts => {
       const voteId = createdVoteId(
         await app.forward(script, { from: holder50 })
       )
-      assert.equal(voteId, 1, 'DotVoting should have been created')
+      assert.equal(getVoteId(voteId), 6, 'DotVoting should have been created')
+    })
+
+    it('unauthorized entities cannot forward to contract', () => {
+      let action = {
+        to: executionTarget.address,
+        calldata: executionTarget.contract.setSignal.getData(
+          [ accounts[7], accounts[8], accounts[9] ],
+          [ 0, 0, 0 ],
+          [ 4, 4, 4 ],
+          'arg1arg2arg3',
+          'description',
+          [ '0x0', '0x0', '0x0' ],
+          [ '0x0', '0x0', '0x0' ],
+          5,
+          false
+        )
+      }
+      const script = encodeCallScript([action])
+      return assertRevert(async () => {
+        await app.forward(script, { from: nonHolder })
+      })
+    })
+
+    it('rejects unequal script length and actual script length', async () => {
```

Comment:
Don't love the test string here, maybe "Rejects if supplied script length and actual script length are not equal"?